pkg/registry/registry.go

```diff
@@ -74,5 +78,18 @@ func New() http.Handler {
 			manifests: map[string]map[string]manifest{},
 		},
 	}
-	return http.HandlerFunc(v.root)
+	for _, o := range opts {
+		o(r)
+	}
+	return http.HandlerFunc(r.root)
+}
+
+// Options describes the available options
+// for creating the registry.
+type Option func(r *registry)
+
+func Logger(l *log.Logger) Option {
```

Comment:
`WithLogger`?

Reply:
I'd prefer `Logger` as it's cleaner.

Reply:
Most options don't seem to add a verb at the front, and it makes them shorter. This also only pollutes the registry namespace which has New(), TLS() and now Logger()